adapters/Mintegral/MintegralAdapter/GADMediationAdapterMintegral.m

```diff
@@ -4,44 +4,98 @@
+#import <MTGSDKBidding/MTGBiddingSDK.h>
 
-- (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
-                         completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  // TODO: Start signal generation and call completionHandler when the signal generation finishes or
-  // fails.
-  completionHandler("", nil);
+
+@implementation GADMediationAdapterMintegral {
+    /// Mintegral banner ad.
+    GADMAdapterMintegralBannerAdLoader *_bannerAd;
+
+    /// Mintegral interstitial ad.
+    GADMAdapterMintegralInterstitialAdLoader *_interstitialAd;
+
+    /// Mintegral native ad.
+    GADMAdapterMintegralNativeAdLoader *_nativeAd;
+
+    /// Mintegral rewarded ad.
+    GADMAdapterMintegralRewardedAdLoader * _rewardedAd;
+}
+
+#pragma mark - GADRTBAdapter
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
+    return [GADMAdapterMintegralExtras class];
 }
 
 + (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
              completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Mintegral SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
+    
+    NSMutableSet *appIds = [[NSMutableSet alloc] init];
+    NSMutableSet *appKeys = [[NSMutableSet alloc] init];
+    
+    for (GADMediationCredentials *credential in configuration.credentials) {
+        NSString *appId = credential.settings[GADMAdapterMintegralAppID];
+        GADMAdapterMintegralMutableSetAddObject(appIds, appId);
+        
+        NSString *appKey =credential.settings[GADMAdapterMintegralAppKey];
+        GADMAdapterMintegralMutableSetAddObject(appKeys, appKey);
+    }
+    
+    if (appIds.count < 1 || appKeys.count < 1) {
+        NSError *error = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters, @"Mintegral mediation configurations did not contain a valid app ID or app Key.");
+        completionHandler(error);
+        return;
+    }
+    
+    NSString *appId = [appIds anyObject];
+    if (appIds.count > 1) {
+        GADMediationAdapterMintegralLog(
+          @"Found the following App IDs:%@. Please remove any app IDs which you are not using.",
+          appIds);
+        GADMediationAdapterMintegralLog(@"Configuring Mintegral SDK with the app ID:%@", appId);
+    }
+    
+    NSString *appKey = [appKeys anyObject];
+    if (appKeys.count > 1) {
+        GADMediationAdapterMintegralLog(
+          @"Found the following App keys:%@. Please remove any App Keys which you are not using.",
+                                        appKeys);
+        GADMediationAdapterMintegralLog(@"Configuring Mintegral SDK with the app key:%@", appKey);
+    }
+    
+    // Initialize the Mintergral SDK.
+    [GADMediationAdapterMintegral setAdmobChannel];
+    [[MTGSDK sharedInstance] setAppID:appId ApiKey:appKey];
```

Comment:
@Mintegral-Lee 

Can the Mintegral SDK handle any combination of `appId`s and `appKey`s ?
If I have two `appID`s and `appKey`s from two different apps configured on the mediation config, as outlined below:
 * My App
   * AppID1
   * AppKey1
 * My Other App 
   * AppID2
   * AppKey2 

How would the Mintegral SDK handle initializing with `AppID1` with `AppKey2` or vice versa? Would init fail? Which between "My App" and "My Other App" would be initialized if it succeeds?

Reply:
When initializing MintegralAdSDK, the MintegralAdSDK will not verify any appID and appKey. If the appID or AppKey is wrong, it will return an error message when loading the ad.

Reply:
Can you add the below comment above this line?
```
Mintegral SDK doesn't verify App ID and App Key when initializing, but will return a failure callback if loading an ad with the incorrect App ID or App Key.
```